vmbindings/dummyvm/src/tests/allocate_align_offset.rs

```diff
@@ -0,0 +1,45 @@
+// GITHUB-CI: MMTK_PLAN=all
+
+use crate::api;
+use crate::DummyVM;
+use crate::tests::fixtures::{Fixture, MutatorFixture};
+use mmtk::plan::AllocationSemantics;
+use mmtk::vm::VMBinding;
+use log::info;
+
+lazy_static! {
+    static ref MUTATOR: Fixture<MutatorFixture> = Fixture::new();
+}
+
+#[test]
+pub fn allocate_alignment() {
+    MUTATOR.with_fixture(|fixture| {
+        let min = DummyVM::MIN_ALIGNMENT;
+        let max = DummyVM::MAX_ALIGNMENT;
+        info!("Allowed alignment between {} and {}", min, max);
+        let mut align = min;
+        while align <= max {
+            info!("Test allocation with alignment {}", align);
+            let addr = api::mmtk_alloc(fixture.mutator, 8, align, 0, AllocationSemantics::Default);
+            assert!(addr.is_aligned_to(align), "Expected allocation alignment {}, returned address is {:?}", align, addr);
+            align *= 2;
+        }
+    })
+}
+
+#[test]
+pub fn allocate_offset() {
+    MUTATOR.with_fixture(|fixture| {
+        const OFFSET: isize = 4;
+        let min = DummyVM::MIN_ALIGNMENT;
+        let max = DummyVM::MAX_ALIGNMENT;
+        info!("Allowed alignment between {} and {}", min, max);
+        let mut align = min;
+        while align <= max {
+            info!("Test allocation with alignment {} and offset {}", align, OFFSET);
+            let addr = api::mmtk_alloc(fixture.mutator, 8, align, OFFSET, AllocationSemantics::Default);
```

Comment:
As I mentioned in a previous comment, adding an offset may require the object to be allocated in a larger size class than the size requested.  In this case, the 8-byte allocation should be allocated in a 16-byte cell because of `OFFSET`, but I suspect it currently isn't.  I recommend add an assertion that the allocated region (from `addr` to `addr + size`) lies within the cell.

There is no public API that exposes that information.  We may add a `debug_assert!` inside the freelist allocator instead of this test case.

Alternatively, we can optionally (e.g. via a feature, maybe named "testing") expose some functions for performing white-box testing, only.

Reply:
I added an assertion in the allocator to check if the allocation is within the cell. It seems working fine with these tests.